src/main/kotlin/ecommerce/configuration/JwtAuthInterceptor.kt

```diff
@@ -0,0 +1,26 @@
+package ecommerce.configuration
+
+import ecommerce.exception.UnauthorizedException
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.stereotype.Component
+import org.springframework.web.servlet.HandlerInterceptor
+
+@Component
+class JwtAuthInterceptor(
+    private val jwtTokenProvider: JwtTokenProvider,
+) : HandlerInterceptor {
+    override fun preHandle(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        handler: Any,
+    ): Boolean {
+        val accessToken = request.getHeader("Authorization")?.substringAfter("Bearer ")
+        if (accessToken == null) throw UnauthorizedException("Invalid Token")
```

Comment:
```suggestion
        val accessToken = request.getHeader("Authorization")?.substringAfter("Bearer ")
            ?: throw UnauthorizedException("Invalid Token")
```

It could be more idiomatic Kotlin 👀

Reply:
I will learn to pay attention to this small details. Thanks for pointing this out :)